internal/sddc/sddc_host_subresource.go

```diff
@@ -0,0 +1,199 @@
+/*
+ *  Copyright 2023 VMware, Inc.
+ *    SPDX-License-Identifier: MPL-2.0
+ */
+
+package sddc
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	utils "github.com/vmware/terraform-provider-vcf/internal/resource_utils"
+	"github.com/vmware/vcf-sdk-go/models"
+)
+
+var portGroup = []string{"VSAN", "VMOTION", "PUBLIC", "MANAGEMENT", "NSX_VTEP", "HOSTMANAGEMENT", "CLOUD_VENDOR_API", "REPLICATION"}
+
+func GetSddcHostSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Required: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"association": {
+					Type:        schema.TypeString,
+					Description: "Host Association: Location/Datacenter",
+					Required:    true,
+				},
+				"credentials": getCredentialsSchema(),
+				"hostname": {
+					Type:         schema.TypeString,
+					Description:  "Host FQDN Example: esx-1, length from 3 to 63",
```

Comment:
"esx-1" is a hostname, not an FQDN. I think `hostname` would be inaccurate.

Reply:
both are supported - if just the short hostname is provided, then FQDN will be generated using the subdomain from dnsSpec

Reply:
So I should not remove hostname and include "if short hostname is provided, then FQDN will be generated using the "domain" attribute from dnsSpec" in the description of the attribute?

Reply:
yes, and this applies to all other hostname fields - vcenter, sddc-manager, nsx, etc.